app/views/users/nbp_wallet/connect.html.slim

```diff
@@ -5,15 +5,21 @@
 
 .row
   .col-8.col-md-4.col-lg-3
-    = link_to @template.url
-      = image_tag @template.qr_code_path, data: {'template-validity': @template.remaining_validity&.seconds}, alt: t('.qr_code_alt_text'), class: 'img-fluid nbp_wallet_qr_code'
-    = button_tag class: 'btn btn-primary regenerate-qr-code-button w-100 mt-3' do
+    = link_to @relationship_template.url
+      = image_tag @relationship_template.qr_code_path,
+        data: {id: 'nbp_wallet_qr_code', 'remaining-validity': @relationship_template.remaining_validity},
+        alt: t('.qr_code_alt_text'),
+        class: 'img-responsive'
```

Comment:
Changing the class might have unintended consequences:

| `img-fluid nbp_wallet_qr_code` | `img-responsive` |
|--------|--------|
| <img width="868" alt="Screenshot 2025-01-30 at 14 09 06" src="https://github.com/user-attachments/assets/67ea0d1f-a9b9-4ce8-be0f-1ef48d673620" /> | <img width="913" alt="Screenshot 2025-01-30 at 14 09 16" src="https://github.com/user-attachments/assets/3e12fa4c-f3cc-4797-a4df-bcab63f3fbef" /> |